src/ConcordiumNetSdk/Transactions/PreparedAccountTransaction.cs

```diff
@@ -0,0 +1,76 @@
+using ConcordiumNetSdk.Types;
+
+namespace ConcordiumNetSdk.Transactions;
+
+/// <summary>
+/// An account transaction which is ready to be signed.
+/// </summary>
+public class PreparedAccountTransaction<T>
+    where T : AccountTransactionPayload<T>
+{
+    /// <summary>
+    /// Address of the sender of the transaction.
+    /// </summary>
+    private AccountAddress _sender;
+
+    /// <summary>
+    /// Account nonce to use for the transaction.
+    /// </summary>
+    private AccountNonce _nonce;
+
+    /// <summary>
+    /// Expiration time of the transaction.
+    /// </summary>
+    private Expiry _expiry;
+
+    /// <summary>
+    /// Payload to send to the node.
+    /// </summary>
+    private AccountTransactionPayload<T> _payload;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PreparedAccountTransaction"/> class.
+    /// </summary>
+    /// <param name="sender">Address of the sender of the transaction.</param>
+    /// <param name="nonce">Account nonce to use for the transaction.</param>
+    /// <param name="expiry">Expiration time of the transaction.</param>
+    /// <param name="payload">Payload to send to the node.</param>
+    private PreparedAccountTransaction(
+        AccountAddress sender,
+        AccountNonce nonce,
+        Expiry expiry,
+        AccountTransactionPayload<T> payload
+    )
+    {
+        this._sender = sender;
+        this._nonce = nonce;
+        this._expiry = expiry;
+        this._payload = payload;
+    }
+
+    /// <summary>
+    /// Creates a new instance of the prepared transaction.
+    /// </summary>
+    /// <param name="sender">Address of the sender of the transaction.</param>
+    /// <param name="nonce">Account nonce to use for the transaction.</param>
+    /// <param name="expiry">Expiration time of the transaction.</param>
+    /// <param name="payload">Payload to send to the node.</param>
+    public static PreparedAccountTransaction<T> Create(
+        AccountAddress sender,
+        AccountNonce nonce,
+        Expiry expiry,
+        AccountTransactionPayload<T> payload
+    )
+    {
+        return new PreparedAccountTransaction<T>(sender, nonce, expiry, payload);
```

Comment:
What is the point of a private and public constructors if the public one just calls the private one without doing anything?